### PR TITLE
Add a 6.10 linux kernel to the virtualized testing in CI

### PR DESCRIPTION
**Pull request by astoycos**
Ensure we run our integration tests against `linux-image-6.10.4-cloud-amd64-unsigned_6.10.4-1_amd64.deb` 
    
Bump xdpilone version to fix bug on latest kernel.

### CODE REVIEW

**Comment by astoycos** [comment]
Blocked on https://github.com/HeroicKatora/xdpilone/issues/3

**Comment by astoycos** [comment]
@Tuetuopay It looks like you added the xdpilone dep wdyt about this failure ^^ I'm seeing it locally on Fedora 40 as well

**Comment by tamird** [comment]
We shouldn't need to bump the host kernel to test on a particular kernel. That's the whole point of the work done in https://github.com/aya-rs/aya/pull/733. Can we just add another kernel to the integration tests?

**Comment by Tuetuopay** [comment]
Seems to be fixed upstream. I’ll take a look tomorrow wrt backwards compatibility, but given the fix it should be good. The integration tests should only need a version bump.

**Comment by astoycos** [comment]
> We shouldn't need to bump the host kernel to test on a particular kernel. That's the whole point of the work done in #733. Can we just add another kernel to the integration tests?

Hrm @tamird  but for the the [`runner.os == 'Linux'` case we still run the integration tests in "local mode"](https://github.com/aya-rs/aya/blob/main/.github/workflows/ci.yml#L284).   I do agree that for the [debian kernels we download](https://github.com/aya-rs/aya/blob/main/.github/workflows/ci.yml#L254) I can just bump the version there.

**Comment by tamird** [comment]
> > We shouldn't need to bump the host kernel to test on a particular kernel. That's the whole point of the work done in #733. Can we just add another kernel to the integration tests?
> 
> Hrm @tamird but for the the [`runner.os == 'Linux'` case we still run the integration tests in "local mode"](https://github.com/aya-rs/aya/blob/main/.github/workflows/ci.yml#L284). I do agree that for the [debian kernels we download](https://github.com/aya-rs/aya/blob/main/.github/workflows/ci.yml#L254) I can just bump the version there.

Yes, and? Unsupported tests should be skipped.

File: .github/workflows/ci.yml
Comment (astoycos) [reply .github/workflows/ci.yml:271]: Ack

Comment (tamird) [reply .github/workflows/ci.yml:271]: Can you undo all the unrelated changes please?

Comment (astoycos) [reply .github/workflows/ci.yml:271]: Done

Comment (tamird) [reply .github/workflows/ci.yml:271]: I meant the loops and arrays - there's no need for this, you can just add lines in the existing formulation

```
          printf '%s\0' \
            linux-image-6.1.0-16-cloud-arm64-unsigned_6.1.67-1_arm64.deb \
            linux-image-6.10.4-cloud-arm64-unsigned_6.10.4-1_arm64.deb \
```
this would work just fine.

**Review by tamird — APPROVED** [review]
Please update the PR title and description before merging.

**Pull request merged**
